x-pack/plugins/maps/public/components/layer_panel/filter_editor/filter_editor.js

```diff
@@ -0,0 +1,200 @@
+  }
+
+  componentWillUnmount() {
+    this._isMounted = false;
+  }
+
+  _loadIndexPatterns = async () => {
+    const indexPatternIds = this.props.layer.getIndexPatternIds();
+    const indexPatterns = [];
+    const getIndexPatternPromises = indexPatternIds.map(async (indexPatternId) => {
+      try {
+        const indexPattern = await indexPatternService.get(indexPatternId);
+        indexPatterns.push(indexPattern);
+      } catch(err) {
+        // unable to fetch index pattern
+      }
+    });
+
+    await Promise.all(getIndexPatternPromises);
+
+    if (!this._isMounted) {
+      return;
+    }
+
+    this.setState({ indexPatterns });
+  }
+
+  _openModal = () => {
+    this.setState({ isModalOpen: true });
+  }
+
+  _closeModal = () => {
+    this.setState({ isModalOpen: false });
+  }
+
+  _onQueryChange = ({ query }) => {
+    this.props.setLayerQuery(this.props.layer.getId(), query);
+    this._closeModal();
+  }
+
+  _renderModal() {
+    if (!this.state.isModalOpen) {
+      return null;
+    }
+
+    const layerQuery = this.props.layer.getQuery();
+
+    return (
+      <EuiOverlayMask>
+        <EuiModal
+          onClose={this._closeModal}
+          maxWidth={false}
+          data-test-subj="mapFilterEditorModal"
+        >
+          <EuiModalHeader>
+            <EuiModalHeaderTitle >
+              <FormattedMessage
+                id="xpack.maps.layerPanel.filterEditor.modal.headerTitle"
+                defaultMessage="Set layer filter"
+              />
+            </EuiModalHeaderTitle>
+          </EuiModalHeader>
+
+          <EuiModalBody className="mapFilterEditorModalBody">
+            <QueryBar
+              query={layerQuery ? layerQuery : { language: settings.get('search:queryLanguage'), query: '' }}
+              onSubmit={this._onQueryChange}
+              appName="maps"
+              showDatePicker={false}
+              indexPatterns={this.state.indexPatterns}
+              store={localStorage}
+              customSubmitButton={
+                <EuiButton
+                  fill
+                  data-test-subj="mapFilterEditorSubmitButton"
+                >
+                  <FormattedMessage
+                    id="xpack.maps.layerPanel.filterEditor.modal.queryBarSubmitButtonLabel"
+                    defaultMessage="Set filter"
+                  />
+                </EuiButton>
+              }
+            />
+          </EuiModalBody>
+
+          <EuiModalFooter>
+            <EuiButtonEmpty
+              onClick={this._closeModal}
+            >
+              <FormattedMessage
+                id="xpack.maps.layerPanel.filterEditor.modal.cancelButtonLabel"
+                defaultMessage="Cancel"
+              />
+            </EuiButtonEmpty>
+
+          </EuiModalFooter>
+        </EuiModal>
+      </EuiOverlayMask>
+    );
+  }
+
+  _renderQuery() {
+    const query = this.props.layer.getQuery();
+    if (!query || !query.query) {
+      return null;
```

Comment:
If there is no query, put in a description of what filters does or why the user would use them.